Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -5,23 +5,45 @@ import NIO
 import NIOFoundationCompat
 import NIOHTTP1
 
+public protocol ToolchainDownloader {
+    func downloadToolchain(
+        _ toolchain: ToolchainVersion,
+        url: String,
+        to destination: String,
+        reportProgress: @escaping (SwiftlyHTTPClient.DownloadProgress) -> Void
+    ) async throws
+}
+
 /// HTTPClient wrapper used for interfacing with various APIs and downloading things.
-public class HTTP {
+public class SwiftlyHTTPClient {
     private static let client = HTTPClientWrapper()
```

Comment:
You create a new HTTPClient with every instance of SwiftlyHTTPClient. Perhaps HTTPClientWrapper should be a wrapper for a global instance.

Reply:
That's how it used to be, but it was changed to be this way so that tests could construct individual commands with their own mocked HTTP clients. The lifetime of a swiftly process is only a single command in actual usage, so having them per-command is essentially the same as a singleton in practice.